src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -51,6 +51,12 @@ public class AvaloniaObject : IAvaloniaObject, IAvaloniaObjectDebug, INotifyProp
         /// </summary>
         private EventHandler<AvaloniaPropertyChangedEventArgs> _propertyChanged;
 
+        /// <summary>
+        /// Delayed setter helper for direct properties. Used to fix #855.
+        /// </summary>
+        private readonly DeferredSetter<AvaloniaProperty, object> directDelayedSetter = new DeferredSetter<AvaloniaProperty, object>();
```

Comment:
I don't really like having a `DeferredSetter` instance for _every_ `AvaloniaObject`... Could we make this static like in `PriorityValue` and key it on the instance and the property or something?

Reply:
I'll mess with that. I just need to make sure we don't cause memory leaks from making it static.

Reply:
The values in the deferred setter are transitory, right? That is, they won't last beyond the bounds of a `SetValue` call? Can this be used to clear up the reference to the object?